src/app/space/add-space-overlay/add-space-overlay.component.ts

```diff
@@ -41,18 +39,10 @@ export class AddSpaceOverlayComponent implements OnInit {
     private notifications: Notifications,
     private userService: UserService,
     private spaceNamespaceService: SpaceNamespaceService,
-    private context: ContextService,
     private broadcaster: Broadcaster
-  ) {
-    this.space = this.createTransientSpace();
-  }
+  ) {}
 
   ngOnInit() {
-    this.subscriptions.push(this.context.current.subscribe((ctx: Context) => {
-      if (ctx.space) {
-        this.currentSpace = ctx.space;
-      }
-    }));
     setTimeout(() => this.spaceNameInput.nativeElement.focus());
```

Comment:
I do not know, this is existing code so maybe @joshuawilson  can answer.

Reply:
`setTimeout` without the time will effectively run the function when the browser is free to run it. It is basically, saying "when you are free, run this code"

I don't know if this was intentional or the reason for it.

Reply:
My point is that when you call the function directly (without `setTimout`), it does exactly what you described "when you are free, run this code", doesn't it?

Reply:
Since JS is single threaded, I understand this to be a bit different. Without the timeout, it gets added to the work done on the thread. With the timeout, it waits for the thread to complete everything else first.

Reply:
still not sure why it is done or if it is a typo and it should have the time added

Reply:
BTW there is a bug reported for this part of code https://github.com/openshiftio/openshift.io/issues/4617

Reply:
Do you see this as a reason to NOT merge this?

Reply:
no, let's merge this PR as soon as possible, I plan to re-run PR checks in ~30min.